### ISSUE TITLE
Getting Started with Flowforge v0.7.0 on macOS; "Editor Unavailable"

### ISSUE THREAD

**Issue by Steveorevo** [flowforge/flowforge#806]
### Current Behavior

Followed the install guide for macOS Big Sur 10.6.8 (Intel i7), Node-RED 3.0.1, Flowforge v0.7.0, NodeJS v16.16.0.

Installation went fine and can bring up http://127.0.0.1:3000 without issue. Creating a first project however perpetually says "editor unavailable", status: starting. Logs appears to hang for the given project. Output window indicates:

```
[2022-07-27T00:23:17.236Z] INFO: Creating project directory: /opt/flowforge/var/projects/08146d13-1f67-4398-978c-907d192323ea
[2022-07-27T00:23:17.336Z] INFO: Set FORGE_NR_PATH to /opt/flowforge/var/stacks/3.0.1
RequestError: connect ECONNREFUSED 127.0.0.1:8880
    at ClientRequest.<anonymous> (/opt/flowforge/app/node_modules/got/dist/source/core/index.js:970:111)
    at Object.onceWrapper (node:events:642:26)
    at ClientRequest.emit (node:events:539:35)
    at ClientRequest.origin.emit (/opt/flowforge/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at Socket.socketErrorListener (node:_http_client:454:9)
    at Socket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1187:16) {
  code: 'ECONNREFUSED',
  timings: {
    start: 1658881401533,
    socket: 1658881401533,
    lookup: 1658881401534,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1658881401534,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 1,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 1
    }
  }
}
[2022-07-27T00:23:41.697Z] INFO: Starting project 08146d13-1f67-4398-978c-907d192323ea on port 7880
[2022-07-27T00:23:41.782Z] INFO: Set FORGE_NR_PATH to /opt/flowforge/var/stacks/3.0.1
[2022-07-27T00:24:41.707Z] INFO: Starting project 08146d13-1f67-4398-978c-907d192323ea on port 7880
[2022-07-27T00:24:41.802Z] INFO: Set FORGE_NR_PATH to /opt/flowforge/var/stacks/3.0.1
[2022-07-27T00:25:41.749Z] INFO: Starting project 08146d13-1f67-4398-978c-907d192323ea on port 7880
...
```

As mentioned, visiting logs for the given project says:

```
[2022-07-27T00:40:09.582Z] ERROR: connect ECONNREFUSED 127.0.0.1:8880
    req: {
      "method": "GET",
      "url": "/api/v1/projects/08146d13-1f67-4398-978c-907d192323ea/logs",
      "hostname": "127.0.0.1:3000",
      "remoteAddress": "127.0.0.1",
      "remotePort": 50925
    }
    res: {
      "statusCode": 500
    }
    reqId: "req-rh"
    err: {
      "type": "RequestError",
      "message": "connect ECONNREFUSED 127.0.0.1:8880",
      "stack":
          RequestError: connect ECONNREFUSED 127.0.0.1:8880
              at ClientRequest.<anonymous> (/opt/flowforge/app/node_modules/got/dist/source/core/index.js:970:111)
              at Object.onceWrapper (node:events:642:26)
              at ClientRequest.emit (node:events:539:35)
              at ClientRequest.origin.emit (/opt/flowforge/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
              at Socket.socketErrorListener (node:_http_client:454:9)
              at Socket.emit (node:events:527:28)
              at emitErrorNT (node:internal/streams/destroy:157:8)
              at emitErrorCloseNT (node:internal/streams/destroy:122:3)
              at processTicksAndRejections (node:internal/process/task_queues:83:21)
              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1187:16)
      "name": "RequestError",
      "code": "ECONNREFUSED",
      "timings": {
        "start": 1658882409581,
        "socket": 1658882409581,
        "lookup": 1658882409582,
        "error": 1658882409582,
        "phases": {
          "wait": 0,
          "dns": 1,
          "total": 1
        }
      }
    }
[2022-07-27T00:40:09.845Z] ERROR: connect ECONNREFUSED 127.0.0.1:8880
    req: {
      "method": "GET",
      "url": "/api/v1/projects/08146d13-1f67-4398-978c-907d192323ea/logs",
      "hostname": "127.0.0.1:3000",
      "remoteAddress": "127.0.0.1",
      "remotePort": 50930
    }
    res: {
      "statusCode": 500
    }
    reqId: "req-rj"
    err: {
      "type": "RequestError",
      "message": "connect ECONNREFUSED 127.0.0.1:8880",
      "stack":
          RequestError: connect ECONNREFUSED 127.0.0.1:8880
              at ClientRequest.<anonymous> (/opt/flowforge/app/node_modules/got/dist/source/core/index.js:970:111)
              at Object.onceWrapper (node:events:642:26)
              at ClientRequest.emit (node:events:539:35)
              at ClientRequest.origin.emit (/opt/flowforge/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
              at Socket.socketErrorListener (node:_http_client:454:9)
              at Socket.emit (node:events:527:28)
              at emitErrorNT (node:internal/streams/destroy:157:8)
              at emitErrorCloseNT (node:internal/streams/destroy:122:3)
              at processTicksAndRejections (node:internal/process/task_queues:83:21)
              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1187:16)
      "name": "RequestError",
      "code": "ECONNREFUSED",
      "timings": {
        "start": 1658882409844,
        "socket": 1658882409845,
        "lookup": 1658882409845,
        "error": 1658882409845,
        "phases": {
          "wait": 1,
          "dns": 0,
          "total": 1
        }
      }
    }
...
```

Wondering if this is a gatekeeper/permissions issue. Terminal is the parent process and I've given that Full Disk Access on macOS in addition to the node stack that's executing (/usr/local/bin/node). 

### Expected Behavior

A link to the project's Node-RED editor instance and a status of "started" or "ready"; haven't used Flowforge before but this is what I'd imagine?

### Steps To Reproduce

Following user guide for local evaluation/installation at:
https://flowforge.com/docs/install/local/

### Environment

- FlowForge version: v0.7.0
- Node.js version: v16.16.0
- npm version: v8.11.0
- Platform/OS: macOS Big Sur 10.6.8 (Intel i7 processor)
- Browser: Firefox 103.0

**Comment by hardillb** [comment]
Under the flowforge install directory there should be a directory called `var/projects`. In here there should be a folder with the name `08146d13-1f67-4398-978c-907d192323ea` This is the home directory for the project you created.

In it there should be a file named `out.log`, could you please attach the content of that file please.

**Comment by Steveorevo** [comment]
Thanks hardillb, here is the output:

```
executable not found
[
  '/opt/flowforge/app/node_modules/@flowforge/nr-launcher/node_modules',
  '/opt/flowforge/app/node_modules/@flowforge/node_modules',
  '/opt/flowforge/app/node_modules',
  '/opt/flowforge/node_modules',
  '/opt/node_modules',
  '/node_modules'
]
executable not found
[
  '/opt/flowforge/app/node_modules/@flowforge/nr-launcher/node_modules',
  '/opt/flowforge/app/node_modules/@flowforge/node_modules',
  '/opt/flowforge/app/node_modules',
  '/opt/flowforge/node_modules',
  '/opt/node_modules',
  '/node_modules'
]
executable not found
[
  '/opt/flowforge/app/node_modules/@flowforge/nr-launcher/node_modules',
  '/opt/flowforge/app/node_modules/@flowforge/node_modules',
  '/opt/flowforge/app/node_modules',
  '/opt/flowforge/node_modules',
  '/opt/node_modules',
  '/node_modules'
]
executable not found
```

It doesn't look like the first path in the array at "/opt/flowforge/app/node_modules/@flowforge/nr-launcher" contains a node_modules folder. I tried running npm install within this folder to obtain dependencies. A node_modules folder now exists but the UI is the same and the error log still looks the same (appended with new entries of the same error and array of paths).

**Comment by hardillb** [comment]
Just checking are you using something like nvm to switch nodejs versions?

**Comment by hardillb** [comment]
Also can we get the output of

`ls -lha /opt/flowforge/app/node_modules/` 

and 

`ls -lha /opt/flowforge/app/node_modules/.bin`

**Comment by Steveorevo** [comment]
Not using nvm; nodejs was installed using the macOS (x64) installer for 16.16.0 LTS from nodejs.org. The requested outputs for attached (list1.txt and list2.txt for the latter .bin). 

[list1.txt](https://github.com/flowforge/flowforge/files/9204363/list1.txt)
[list2.txt](https://github.com/flowforge/flowforge/files/9204364/list2.txt)

**Comment by hardillb** [comment]
I think I've worked out what is up here.

Can you look in the `/opt/flowforge/var` directory and check if there is a `stacks` directory?

In the `stacks` directory there should be a directory that matches the version of Node-RED that you defined in your stack. (e.g. `2.2.2` or `3.0.1`).

Stacks are installed with the `install-stack` command (docs here: https://flowforge.com/docs/install/local/stacks/)

You need to both define the stack in the UI and install the Node-RED version for the stack on the command line and when asked for the NR version in the UI it must match exactly (e.g. `2.2.2` not `v2.2.2`) the version passed to the `install-stack` command.

Th

**Comment by Steveorevo** [comment]
Success! Yes, this is indeed the case; there was no stacks directory. I followed the suggested docs and was able to get Node-RED up and running with one, slight modification. The indicator is now green in FlowForge and says "running" with a proper link to the editor; which works beautifully. I used the following command to get 3.0.1 installed

```
cd /opt/flowforge/bin
./ff-install-stack.sh 3.0.1
```

This coordinated with the stack I defined.

**Q:** The command to install the stack is pre-fixed with `ff-`, and appears to be true for all platforms. Is this true for development and production? The former being corrected to also be prefixed with `ff-`:

```
bash
npm run ff-install-stack --vers=2.2.2
```
I've created a pull request to update documentation accordingly. 

Thank you for your help in this matter!
